File: client/backend/src/main/java/com/eduquestia/backend/dto/request/CompletarMisionRequest.java
package com.eduquestia.backend.dto.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CompletarMisionRequest {
    
    @NotBlank(message = "El contenido de la entrega es obligatorio")
    private String contenidoEntrega;
    
    private String archivoUrl;
    
    private String comentariosEstudiante;
}

File: client/frontend-mobile/app/src/main/java/com/eduquestia/frontend_mobile/data/model/CompletarMisionRequest.kt
package com.eduquestia.frontend_mobile.data.model

import kotlinx.serialization.Serializable

/**
 * Request para completar una misión
 */
@Serializable
data class CompletarMisionRequest(
    val contenidoEntrega: String,
    val archivoUrl: String? = null,
    val comentariosEstudiante: String? = null
)

